### PR TITLE
NAPI_Initialize NAPI_Finalize test_AES128

### PR DESCRIPTION
**Pull request by Liang1XZhao**
c++ complete NAPI_Initialize 、NAPI_Finalize、 test_AES128
node-service add NAPI_Initialize、NAPI_Finalize

### CODE REVIEW

File: core/App/ehsm_napi.cpp [hunk 1]
```diff
@@ -36,53 +36,37 @@
...
-        retString += "}";
-		return StringToChar(retString);
-	};
-} RetJsonObj;
 
-extern "C" {
+/*
+destory the enclave
+*/
+void NAPI_Finalize(){
```
Comment (yang8621) [inline core/App/ehsm_napi.cpp:67]: not found the correct place to call?

Comment (Liang1XZhao) [reply core/App/ehsm_napi.cpp:67]: This method is called in nodejs.

Comment (yang8621) [reply core/App/ehsm_napi.cpp:67]: I only find ehsm_napi.NAPI_Initialize() is called.   Where is NAPI_Finalize()?

Comment (Liang1XZhao) [reply core/App/ehsm_napi.cpp:67]: It will be fix.

File: core/App/ehsm_core_test.cpp [hunk 2]
```diff
@@ -42,577 +42,60 @@
...
-    SAFE_FREE(plaintext.data);
-    SAFE_FREE(ciphertext.data);
-    SAFE_FREE(plaintext2.data);
-
-    printf("============testAES Done==========\n");
-
-    return ret;
-}
-
-ehsm_status_t testRSA()
```
Comment (syan10) [inline core/App/ehsm_core_test.cpp:169]: don't delete the original test cases unless you have already implemented the new one.

File: core/App/ehsm_core_test.cpp [hunk 3]
```diff
@@ -42,577 +42,60 @@
...
-    SAFE_FREE(cipher_datakey.data);
-    SAFE_FREE(cipher_datakey2.data);
-    SAFE_FREE(plaintext1.data);
-    SAFE_FREE(plaintext2.data);
-
-    printf("============testGenerateDataKey done==========\n");
-    return ret;
+    
+out:
+    printf("============test_AES128 end==========\n");
```
Comment (syan10) [inline core/App/ehsm_core_test.cpp:96]: seems that you didn't free the memory!！

the NAPI_xxx function will return a json string, i'm not sure whether the nodejs side will free them automatically? and for the c++ unittest, it seems that you need to free them by yourself.

**Review by syan10 — CHANGES_REQUESTED** [review]
-1

File: core/App/ehsm_core_test.cpp (hunk 3 above)
Comment (Liang1XZhao) [reply core/App/ehsm_core_test.cpp:96]: It will be fix

**Comment by Liang1XZhao** [comment]
resubmit after repair

**Pull request closed without merging**